feedhenry/fh-ios-sdk
# drop JSONKit dependency

**Motivation:**
The sdk uses the old/deprecated JSONKit dependency. From iOS 5+ and later, the platform provides ```NSJSONSerialization``` class to perform JSON parsing. This PR adds a new class [FHJSON](https://github.com/cvasilak/fh-ios-sdk/blob/FHMOBSDK-42.drop-jsonkit/fh-ios-sdk/Categories/JSON/FHJSON.h)  with categories to existing collection foundation classes which invokes the ```NSJSONSerialization``` internally.
Care was taken to follow the same convenient method name's used by JSONKit, so the changes to the library level plus user application's code are ```minimal```, e.g simple replacement of ```#import "JSONKit.h"``` with ```#import "FHJSON.h"```

Further, since many demos use those convenient methods, have adjusted the ```Documentation``` target to generate them as public visible (added appledoc), so we are explicit that they can be used in user's application land. Eg. look at the [Category References](http://cvasilak.github.io/fh-ios-sdk/docset/Contents/Resources/Documents/index.html) section in the generated html documentation.

**To test:**
- Simple open the ```fh-ios-sdk.xcodeproj``` and run Product->Tests. The tests should build and run successfully.
- Invoke the ```build.sh``` script and once the fwk is build, open each demo in the /demos directory. They should work as expected (Note: that you need to adjust ```fhconfig.plist``` with your cloud FH app's id etc).

@wei-lee mind to review?
